src/Simplify.elm

```diff
@@ -6370,6 +6379,79 @@ indexAccessChecks collection checkInfo n =
                 Nothing
 
 
+setChecks : CollectionProperties (FromListProperties (IndexableProperties {})) -> CheckInfo -> Maybe (Error {})
+setChecks collection checkInfo =
+    firstThatConstructsJust
+        [ \() ->
+            Maybe.andThen
+                (\emptiableArg -> callOnEmptyReturnsEmptyCheck emptiableArg collection checkInfo)
+                (thirdArg checkInfo)
+        , \() ->
+            case Evaluate.getInt checkInfo checkInfo.firstArg of
+                Just n ->
+                    if n < 0 then
+                        Just
+                            (alwaysReturnsLastArgError
+                                (qualifiedToString checkInfo.fn ++ " with negative index")
+                                collection
+                                checkInfo
+                            )
+
+                    else
+                        case secondArg checkInfo of
+                            Just replacementArg ->
+                                setOnKnownElementChecks collection checkInfo n (Node.range replacementArg)
+
+                            Nothing ->
+                                Nothing
+
+                Nothing ->
+                    Nothing
+        ]
+        ()
+
+
+setOnKnownElementChecks :
+    CollectionProperties (FromListProperties (IndexableProperties {}))
+    -> CheckInfo
+    -> Int
+    -> Range
+    -> Maybe (Error {})
+setOnKnownElementChecks collection checkInfo n replacementArgRange =
+    case thirdArg checkInfo of
+        Just collectionArg ->
+            case collection.literalElements checkInfo.lookupTable collectionArg of
+                Just literalElements ->
+                    case List.drop n literalElements |> List.head of
+                        Just element ->
+                            Just
+                                (Rule.errorWithFix
+                                    { message = qualifiedToString checkInfo.fn ++ " will replace a known element in a literal " ++ collection.represents
+                                    , details = [ "You can move the replacement argument directly into the " ++ collection.represents ++ "." ]
```

Comment:
Instead of using `collection.represents` in both places, why not use `collection.fromListLiteralDescription`?